code/game/objects/items/stacks/sheets/glass.dm

```diff
@@ -329,7 +329,7 @@
 		playsound(loc, 'sound/effects/glass_step.ogg', 50, 1)
 		if(ishuman(AM))
 			var/mob/living/carbon/human/H = AM
-			if(!(PIERCEIMMUNE in H.dna.species.specflags))
+			if(!(PIERCEIMMUNE in H.dna.species.specflags) || isobj(usr.buckled) && !usr.buckled.anchored)
```

Comment:
I think you'll wanted to wrap the `isobj(usr.buckled) && !usr.buckled.anchored` in ( and ) so it goes like this:

* Is the human immune to pierce?
OR
* is the user buckled to something AND is the buckled item not anchored